xarray/core/ops.py

```diff
@@ -140,6 +140,30 @@ def array_equiv(arr1, arr2):
     return bool(flag_array.all())
 
 
+def array_notnull_equiv(arr1, arr2):
+    """Like np.array_equal, but also allows values to be NaN in either or both
+    arrays
+    """
+    arr1, arr2 = as_like_arrays(arr1, arr2)
+    if arr1.shape != arr2.shape:
+        return False
+
+    flag_array = (arr1 == arr2)
+
+    # GH837, GH861
```

Comment:
I'm not sure why we missed this before, but these comments (and ignoring TypeError) should probably go in the `isnull` function instead, rather than repeating it here.